src/domain/shared/utils/useSubspaceCreation/useSubspaceCreation.ts

```diff
@@ -133,14 +156,59 @@ export const useSubspaceCreation = () => {
                 mode: SpacePrivacyMode.Public,
               },
             },
+            visuals: {
+              id: '',
+              cardBanner: {
+                id: '',
+              },
+              avatar: {
+                id: '',
+              },
+            },
           },
         },
       });
-
+      try {
+        const uploadPromises: Promise<unknown>[] = [];
+        if (value.visuals.avatar.file && data?.createSubspace.visuals.avatar?.id) {
+          uploadPromises.push(
+            uploadVisual({
+              variables: {
+                file: value.visuals.avatar.file,
+                uploadData: {
+                  visualID: data.createSubspace.visuals.avatar.id,
+                  alternativeText: value.visuals.avatar.altText,
+                },
+              },
+            })
+          );
+        }
+        if (value.visuals.cardBanner.file && data?.createSubspace.visuals.cardBanner?.id) {
+          uploadPromises.push(
+            uploadVisual({
+              variables: {
+                file: value.visuals.cardBanner.file,
+                uploadData: {
+                  visualID: data.createSubspace.visuals.cardBanner.id,
+                  alternativeText: value.visuals.cardBanner.altText,
+                },
+              },
+            })
+          );
+        }
+        await Promise.all(uploadPromises);
+      } catch (error) {
+        // Subspace is created anyway, just the images failed log the error and continue
+        if (error instanceof Error) {
+          logError(error);
```

Comment:
Add suggest labeling the error related to the Temporary Storage, for ease of search and detection.
e.g logError(error, { LABEL: 'TempStorage'});